trieste/models/gpflow/models.py

```diff
@@ -655,18 +581,17 @@ def __init__(
                 )
 
         self._inducing_point_selector = inducing_point_selector
+        self._ensure_variable_model_data()
+        self.create_posterior_cache()
 
-        # GPflow stores num_data as a number. However, since we want to be able to update it
-        # without having to retrace the acquisition functions, put it in a Variable instead.
-        # So that the elbo method doesn't fail we also need to turn it into a property.
-        if not isinstance(self._model, NumDataPropertyMixin):
-
-            class SVGPWrapper(type(self._model), NumDataPropertyMixin):  # type: ignore
-                """A wrapper around GPFlow's SVGP class that stores num_data in a tf.Variable and
-                exposes it as a property."""
-
-            self._model._num_data = tf.Variable(model.num_data, trainable=False, dtype=tf.float64)
-            self._model.__class__ = SVGPWrapper
+    def _ensure_variable_model_data(self) -> None:
+        # GPflow stores the data in Tensors. However, since we want to be able to update the data
+        # without having to retrace the acquisition functions, put it in Variables instead.
+        # Data has to be stored in variables with dynamic shape to allow for changes
+        # Sometimes, for instance after serialization-deserialization, the shape can be overridden
+        # Thus here we ensure data is stored in dynamic shape Variables
+        if not is_variable(self._model.num_data):
+            self._model.num_data = tf.Variable(self._model.num_data, trainable=False)
```

Comment:
Do you actually need this? I think I came to the conclusion it wasn't used anywhere? Or is it used outside `Trieste`?

Reply:
We update `num_data` in `SparseVariational.update` so need to ensure it's in a variable, no?

Reply:
> We update `num_data` in `SparseVariational.update` so need to ensure it's in a variable, no?

But why do you update it? Do you ever read it? Can you not just leave it be?

Reply:
Good question. I vaguely remember it being used internally, but looking at the code it doesn't seem to make sense to override it: if we want it to vary with the data size we would just leave it as None.

I'll remove this and see if anything breaks :-)